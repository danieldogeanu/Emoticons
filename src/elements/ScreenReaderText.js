import React from 'react';

const ScreenReaderText = (props) => (
	<span data-testid="ScreenReaderText"
		className="show-for-screen-reader">
		{props.text}
	</span>
);

export default ScreenReaderText;
