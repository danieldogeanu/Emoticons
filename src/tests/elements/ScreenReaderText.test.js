import React from 'react';
import {render} from '@testing-library/react';
import ScreenReaderText from '../../elements/ScreenReaderText';

describe('ScreenReaderText Element', () => {

	it('renders screen reader text properly', () => {
		const testText = 'Some Test Text';
		const {getByTestId} = render(<ScreenReaderText text={testText} />);
		const renderedSRT = getByTestId('ScreenReaderText');

		expect(renderedSRT).toBeInTheDocument();
		expect(renderedSRT).toHaveClass('show-for-screen-reader');
		expect(renderedSRT).toHaveTextContent(testText);
		expect(renderedSRT).toMatchSnapshot();
	});

});
