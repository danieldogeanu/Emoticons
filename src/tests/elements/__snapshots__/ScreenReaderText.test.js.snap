// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ScreenReaderText Element renders screen reader text properly 1`] = `
<span
  class="show-for-screen-reader"
  data-testid="ScreenReaderText"
>
  Some Test Text
</span>
`;
